modules/frontend/pipeline/deny_list_middleware.go

```diff
@@ -0,0 +1,52 @@
+package pipeline
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/grafana/tempo/modules/frontend/combiner"
+)
+
+type urlDenylistWare struct {
+	denyList []*regexp.Regexp
+	next     AsyncRoundTripper[combiner.PipelineResponse]
+}
+
+func NewURLDenyListWare(denyList []string) AsyncMiddleware[combiner.PipelineResponse] {
+	compiledDenylist := make([]*regexp.Regexp, 0)
+	for _, v := range denyList {
+		r, err := regexp.Compile(v)
+		if err == nil {
+			compiledDenylist = append(compiledDenylist, r)
+		} else {
+			panic(fmt.Sprintf("error compiling query frontend deny list regex: %s", err))
+		}
+	}
+
+	return AsyncMiddlewareFunc[combiner.PipelineResponse](func(next AsyncRoundTripper[combiner.PipelineResponse]) AsyncRoundTripper[combiner.PipelineResponse] {
+		return &urlDenylistWare{
+			next:     next,
+			denyList: compiledDenylist,
+		}
+	})
+}
+
+func (c urlDenylistWare) RoundTrip(req Request) (Responses[combiner.PipelineResponse], error) {
+	if len(c.denyList) != 0 {
+		err := c.validateRequest(req.HTTPRequest().URL.Path)
```

Comment:
i don't believe path includes the url parameters. we need to test against something that includes these params so we can block specific traceql queries

Reply:
right, I thought this was the whole URL